event_routing_backends/processors/mixins/base_transformer.py

```diff
@@ -102,3 +102,16 @@ def transform(self):
                 )
 
         return self.transformed_event
+
+    def extract_username(self):
+        """
+        extract username edX event.
```

Comment:
`Extracts username from event by finding it in context first and falling back to data if context does have username key`